components/style/properties/declaration_block.rs

```diff
@@ -289,9 +289,13 @@ impl PropertyDeclarationBlock {
         let important_count = &mut self.important_count;
         let mut removed_at_least_one = false;
         self.declarations.retain(|&(ref declaration, importance)| {
-            let remove = declaration.id().is_or_is_longhand_of(property);
+            let id = declaration.id();
+            let remove = id.is_or_is_longhand_of(property);
             if remove {
                 removed_at_least_one = true;
+                if let PropertyId::Longhand(id) = *property {
```

Comment:
As discussed on IRC, this is wrong and needs to look at `id`, not at `property`.

r=me with this fixed.